public/bitemp_graph_helper.js

```diff
@@ -134,11 +134,71 @@ function changeTextInGraph(chart, params) {
   if (docProp === '') {
     window.alert('Please enter a document property.');
   }
-  else {
+  var propExists = false;
+
+  for(var i = 0; i < params.data.length && !propExists; i++) {
+    for(var prop in params.data[i].content) {
+      if (params.data[i].content.hasOwnProperty(prop)) {
+        if(prop === docProp) {
+          propExists = true;            
+        }
+      }
+    }
+  }
+  if(propExists) {
+    drawChart(params, docProp);
     getBarChart(params, docProp);
   }
+  else {
+    if(docProp !== '')	{
+      window.alert('Sorry. That property does not exist in any document in the collection');
+    }
+  }
 }
 
+function addDataToMenu(chart, params) {
+  $('#select-prop').empty();
+  if(!params) {
+  	return;
+  }
+  var propsInGraph = [];
+  propsInGraph.push('Choose a property');  
+  var alreadyIn = false;
+/*
+  triple for loop loops through 
+  1.) the data array
+  2.) the properties contained in data
+  3.) the newly created propsInGraph to avoid pushing the same props
+*/
+  for(var i = 0; i < params.data.length; i++) {
```

Comment:
rather than an array, use an object. If you assign true to the same key twice, that's okay. No need to iterate through the array.